x-pack/plugins/infra/public/pages/logs/settings/indices_configuration_panel.tsx

```diff
@@ -25,35 +27,100 @@ export const IndicesConfigurationPanel = React.memo<{
 }>(({ isLoading, isReadOnly, indicesFormElement }) => {
   const trackSwitchToIndexPatternReference = useUiTracker({ app: 'infra_logs' });
 
-  const switchToIndexPatternReference = useCallback(() => {
-    indicesFormElement.updateValue(() => undefined);
-    trackSwitchToIndexPatternReference({
-      metric: 'configuration_switch_to_index_pattern_reference',
-    });
-  }, [indicesFormElement, trackSwitchToIndexPatternReference]);
+  return (
+    <EuiFormFieldset
+      legend={{
+        children: (
+          <EuiTitle size="s">
+            <h3>
+              <FormattedMessage
+                id="xpack.infra.logSourceConfiguration.logSourcesTitle"
+                defaultMessage="Log sources"
+              />
+            </h3>
+          </EuiTitle>
+        ),
+      }}
+    >
+      <EuiCheckableCard
+        id="dataView"
+        label={
+          <EuiTitle size="xs">
+            <h2>
+              <FormattedMessage
+                id="xpack.infra.logSourceConfiguration.dataViewSectionTitle"
+                defaultMessage="Data view (recommended)"
+              />
+            </h2>
+          </EuiTitle>
+        }
+        name="dataView"
+        value="dataView"
+        checked={isIndexPatternFormElement(indicesFormElement)}
+        onChange={() => {
```

Comment:
TL;DR: Yes, this only gets called when the radio is selected.

Details:
Within the `EuiCheckableCard` they render a `EuiRadio` which renders a native `input type="radio"`.  `onChange` is a bit misleading, unlike a `input type="checkbox"`, you cannot deselect a radio button. So the only change possible is "selected".

So while the event to listen to is `change` there is no native event fired for "deselect" because you cannot deselect a single radio button through the UI (it can be done via code, yet no event is fired). Radios are special in that they mainly make sense in a group, the way to "deselect" one is to select another.